doc/embedded/crosscompile_armv7.md

```diff
@@ -0,0 +1,188 @@
+compiled programs.  However, in this tutorial, we will stick to glibc for
+simplicity.
+
+If you want to get a link directly to the toolchain and download it with `wget`
+use the following command:
+
+```sh
+wget https://toolchains.bootlin.com/downloads/releases/toolchains/armv7-eabihf/tarballs/armv7-eabihf--glibc--stable-2024.02-1.tar.bz2
+```
+
+Once we have downloaded the toolchain we can extract the content using the
+following command:
+
+```sh
+tar -xvf armv7-eabihf--glibc--stable-2024.02-1.tar.bz2
+```
+
+Once we have done that, now the toolchain is ready.  Feel free to explore inside
+the toolchain; you will find the basic set of compilers in addition to a
+`sysroot/` folder that contains headers for standard libraries.
+
+Now, we will use this toolchain to compile mlpack. The first thing is to get the
+mlpack sources.  You can either download the archive file of the latest version,
+or clone the repository if you prefer to use the development version.  In this
+case we will clone the repository:
+
+```sh
+git clone git@github.com:mlpack/mlpack.git
+```
+
+Once we have downloaded the source code, now we need to get mlpack external
+dependencies.  mlpack has three dependencies:
+
+ * [ensmallen](https://www.ensmallen.org/), which provides mathematical solvers and optimizers for training;
+ * [cereal](https://uscilab.github.io/cereal/) for model serialization; and
+ * [Armadillo](https://arma.sourceforge.net/) for linear algebra and matrix
+   manipulation.
+
+Armadillo itself can use several backends.  In this tutorials we will use
+OpenBLAS since it is optimized on a broad range of different architectures.
+
+To get the dependencies we have two solutions: either we download them manually,
+or we use the autodownloader.  The three direct dependencies
+(ensmallen/cereal/Armadillo) are all header-only and thus it suffices to simply
+download and unpack the sources, but Armadillo's backend (the BLAS/LAPACK
+implementation) generally is not and so it will need to be cross-compiled for
+the target.
+
+In this tutorial we use the autodownloader since it automates the entire
+process, including the cross-compilation of OpenBLAS.  The first step is to
+create a build directory, just like the
+[regular build process](../../README.md#3-installing-and-using-mlpack-in-c):
+
+```sh
+cd mlpack/
+mkdir build
+cd build
+```
+
+The next step is to use CMake to configure the build.  In this command we are
+telling CMake to use a specific compiler in
+addition to pointing out the sysroot that will indicate where the headers and
+standard C++ library are.  When cross-compiling, we cannot use the host system
+headers or try to link against libraries built for the host.  We specify the
+board name to be `RPI2` so that mlpack will configure the compilation options
+specifically for the Raspberry Pi 2. Other options for the board name are
+available---for instance, `CORTEXA7` could be used too.
```

Comment:
Just a note, we should cross-link to the table in #3778 here (which PR to use for that I guess depends on which gets merged first :smile:)